vizro-core/src/vizro/models/_components/graph.py

```diff
@@ -57,28 +58,16 @@ def __call__(self, **kwargs):
         fig = self.figure(**kwargs)
         fig = self._optimise_fig_layout_for_dashboard(fig)
 
-        # Apply the template vizro_dark or vizro_light by setting fig.layout.template. This is exactly the same as
-        # what the clientside update_graph_theme callback does, and it would be nice if we could just use that by
-        # including Input(self.id, "figure") as input for that callback, but doing so leads to a small flicker between
-        # completion of this serverside callback and starting that clientside callbacks.
-        # Note that this does not fully set the template for plotly.express figures. Doing this post-fig creation update
-        # relies on the fact that we have already set pio.templates.default before the self.figure call
-        # above (but not with the _pio_templates_default context manager surrounding the above self.figure call,
-        # since that would alter global state).
-        # Possibly we should pass through the theme selector as an argument `template` in __call__ rather than fetching
-        # it from ctx here. Remember that passing it as self.figure(template) is not helpful though, because custom
-        # graph figures don't need a template argument, and the clientside them selector callback would override this
-        # anyway.
         # Possibly we should enforce that __call__ can only be used within the context of a callback, but it's easy
         # to just swallow up the error here as it doesn't cause any problems.
-        try:
-            # At the moment theme_selector is always present so this if statement is redundant, but possibly in
-            # future we'll have callbacks that do Graph.__call__() without theme_selector set.
-            if "theme_selector" in ctx.args_grouping.get("external", {}):
-                theme_selector_checked = ctx.args_grouping["external"]["theme_selector"]["value"]
-                fig.layout.template = "vizro_light" if theme_selector_checked else "vizro_dark"
-        except MissingCallbackContextException:
-            logger.info("fig.update_layout called outside of callback context.")
+        with suppress(MissingCallbackContextException):
```

Comment:
Does set_props throw MissingCallbackContextException if the ctx doesn't exist?

Reply:
Yes:
```
dash.exceptions.MissingCallbackContextException: dash.callback_context.set_props is only available from a callback!
```